tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py

```diff
@@ -0,0 +1,460 @@
+            }
+        }
+    }
+    """
+
+    def __init__(self, scheduler, os, cluster, data_retriever):
+        """Get the state of the cluster as it pertains to the CloudWatch logging feature."""
+        self.scheduler = scheduler
+        self.platform = self._base_os_to_platform(os)
+        self.cluster = cluster
+        self.remote_command_executor = RemoteCommandExecutor(self.cluster)
+        self.scheduler_commands = get_scheduler_commands(self.scheduler, self.remote_command_executor)
+        self.data_retriever = data_retriever
+        self._relevant_logs = {"MasterServer": [], "ComputeFleet": []}
+        self._cluster_log_state = {"MasterServer": {}, "ComputeFleet": {}}
+        self._set_cluster_log_state()
+
+    def get_logs_state(self):
+        """Get the state of the log files applicable to each of the cluster's EC2 instances."""
+        desired_keys = ["hostname", "instance_id", "node_role", "agent_status", "logs"]
+        states = [{key: self._cluster_log_state["MasterServer"][key] for key in desired_keys}]
+        states.extend(
+            [
+                {key: host_dict[key] for key in desired_keys}
+                for hostname, host_dict in self._cluster_log_state["ComputeFleet"].items()
+            ]
+        )
+        assert_that(states).is_length(self.scheduler_commands.compute_nodes_count() + 1)  # computes + master
+        return states
+
+    def _dump_cluster_log_state(self):
+        """Dump the JSON-ified string of self._cluster_log_state for debugging purposes."""
+        return json.dumps(self._cluster_log_state, indent=4)
+
+    @staticmethod
+    def _base_os_to_platform(base_os):
+        """Turn the name of a base OS into the platform."""
+        # Special case: alinux is how the config file refers to amazon linux, but in the chef cookbook
+        # (and the cloudwatch log config produced by it) the platform is "amazon".
+        translations = {"alinux": "amazon"}
+        no_digits = base_os.rstrip(string.digits)
+        return translations.get(no_digits, no_digits)
+
+    def _set_master_instance(self, instance):
+        """Set the master instance field in self.cluster_log_state."""
+        self._cluster_log_state["MasterServer"].update(
+            {"node_role": "MasterServer", "hostname": instance["PrivateDnsName"], "instance_id": instance["InstanceId"]}
+        )
+
+    def _add_compute_instance(self, instance):
+        """Update the cluster's log state by adding a compute node."""
+        self._cluster_log_state["ComputeFleet"][instance["PrivateDnsName"]] = {
+            "node_role": "ComputeFleet",
+            "hostname": instance["PrivateDnsName"],
+            "instance_id": instance["InstanceId"],
+        }
+
+    def _get_initial_cluster_log_state(self):
+        """Get EC2 instances belonging to this cluster. Figure out their roles in the cluster."""
+        for reservation in self.data_retriever.get_ec2_instances()["Reservations"]:
```

Comment:
You may need to paginate this call. If the number of instances is high, this could lead to not getting all the instances in the cluster. See https://github.com/aws/aws-parallelcluster/blob/f842e47fa5dca64e98c44e932ba549605f35e345/cli/pcluster/utils.py#L49

Reply:
Done.